extensions/infinispan-cache/runtime/src/main/java/io/quarkus/cache/infinispan/runtime/InfinispanCacheImpl.java

```diff
@@ -0,0 +1,215 @@
+                        computationResults.remove(key);
+                    });
+            return resultAsync;
+        }));
+    }
+
+    @Override
+    public <K, V> Uni<V> getAsync(K key, Function<K, Uni<V>> valueLoader) {
+        return Uni.createFrom().completionStage(handleAndCompose(remoteCache.getAsync(key), (v1, ex1) -> {
+            if (ex1 != null) {
+                return CompletableFuture.failedFuture(ex1);
+            }
+
+            if (v1 != null) {
+                return CompletableFuture.completedFuture(decodeNull(v1));
+            }
+
+            CompletableFuture<V> resultAsync = new CompletableFuture<>();
+            CompletableFuture<V> computedValue = computationResults.putIfAbsent(key, resultAsync);
+            if (computedValue != null) {
+                return computedValue;
+            }
+            valueLoader.apply(key).convert().toCompletionStage()
+                    .whenComplete((newValue, ex2) -> {
+                        if (ex2 != null) {
+                            resultAsync.completeExceptionally(ex2);
+                            computationResults.remove(key);
+                        } else {
+                            remoteCache.putIfAbsentAsync(key, encodeNull(newValue), lifespan, TimeUnit.MILLISECONDS, maxIdle,
+                                    TimeUnit.MILLISECONDS).whenComplete((existing, ex3) -> {
+                                        if (ex3 != null) {
+                                            resultAsync.completeExceptionally((Throwable) ex3);
+                                        } else if (existing == null) {
+                                            resultAsync.complete(newValue);
+                                        } else {
+                                            resultAsync.complete(decodeNull(existing));
+                                        }
+                                        computationResults.remove(key);
+                                    });
+                        }
+                    });
+            return resultAsync;
+        }));
+    }
+
+    @Override
+    public Uni<Void> invalidate(Object key) {
+        return Uni.createFrom().completionStage(remoteCache.removeAsync(key));
+    }
+
+    @Override
+    public Uni<Void> invalidateAll() {
+        return Uni.createFrom().completionStage(remoteCache.clearAsync());
+    }
+
+    @Override
+    public Uni<Void> invalidateIf(Predicate<Object> predicate) {
+        Flow.Publisher<Map.Entry> entriesPublisher = FlowAdapters
+                .toFlowPublisher(remoteCache.publishEntries(Codec27.EMPTY_VALUE_CONVERTER, null, null, 512));
+        return Uni.createFrom().multi(Multi.createFrom().publisher(entriesPublisher)
+                .map(e -> ((Map.Entry<Object, Object>) e).getKey())
+                .filter(key -> predicate.test(key))
+                .onItem()
+                .call(key -> Uni.createFrom().completionStage(remoteCache.removeAsync(key))))
```

Comment:
I thing this can be simplify by merging the `call` and the `map`

Reply:
You can replace the map(r -> null) with .replaceWithNull().

Reply:
I added replaceWithVoid()